### ISSUE TITLE
Network Error for Email Pop-up from within Cases

### ISSUE THREAD

**Issue by JVeitia** [civicrm/org.civicrm.casetokens#2]
This extension causes an error message when attempting to open the email pop-up from within the cases module.  Here's a snapshot of where the error is occurring: 

![image](https://user-images.githubusercontent.com/12940400/39306579-8dbd1cc8-492e-11e8-91a0-ff066ede89e7.png)

I checked the error log and this is what is showed: 
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
 in /home/lofcenter/www/www/wp-content/plugins/civicrm/civicrm/CRM/Core/Error/Log.php on line 69
[26-Apr-2018 08:46:35 America/New_York] PHP Deprecated:  Deprecated function, use CRM_Core_PseudoConstant::getKey
Array
(
    [civi.tag] => deprecated
)
The error went away once I disabled this extension.

**Comment by colemanw** [comment]
Thanks for posting the detailed bug report. I'm afraid that error log doesn't tell me much. It's just minor complaints about non-fatal code quality issues. Nothing that would stop the popup from appearing. Could you please open up your console (F12) and try opening the popup. You should see an XHR request in the console when you do that, and it should turn red or otherwise indicate that it failed. Click on it to get more details about it and post the screenshot. Thanks!

**Comment by shahrukh-compuco** [comment]
@colemanw its is coming from bulk actions on the case list page, to reproduce it select multiple cases and then select “Print/Merge Document” it will throw an error which seems to be coming from https://github.com/civicrm/org.civicrm.casetokens/blob/master/casetokens.php#L118 as we get comma separated case ids so this api errors out.

**Comment by shahrukh-compuco** [comment]
@colemanw any update on this one?

**Comment by colemanw** [comment]
@shahrukh-compuco sounds like the api call needs to be updated to fetch multiple cases? Do you want to do a PR to change it to use Api4 and the IN operator?

**Comment by shahrukh-compuco** [comment]
but @colemanw how would the tokens be resolved in case of multiple cases being selected

**Comment by colemanw** [comment]
> how would the tokens be resolved in case of multiple cases being selected

Good question! Typically tokens that contain multiple values will list them separated by commas. So we could do that: gather the values into an array and then `implode(', ')` them.

**Comment by shahrukh-compuco** [comment]
@colemanw I think this will result in very uncertain scenarios as different cases can have different case roles depending on the case types and also the way the code is structured currently do you think it is possible to support multiple cases or would it make sense to allow these tokens only for single case scenarios.

Please let me know your thoughts so that I can start the work on this

**Comment by colemanw** [comment]
@shahrukh-compuco yes that's probably the simplest fix, just use the first case id when resolving the token.

**Comment by shahrukh-compuco** [comment]
@colemanw here is the PR https://github.com/civicrm/org.civicrm.casetokens/pull/12 can you please review and merge it
